src/gui/systray.cpp

```diff
@@ -193,6 +194,37 @@ void Systray::setPauseOnAllFoldersHelper(bool pause)
     }
 }
 
+void Systray::slotForceSyncAllFolders()
+{
+    // For some reason we get the raw pointer from Folder::accountState()
+    // that's why we need a list of raw pointers for the call to contains
+    // later on...
+    const auto accounts = [=] {
+        const auto ptrList = AccountManager::instance()->accounts();
+        auto result = QList<AccountState *>();
+        result.reserve(ptrList.size());
+        std::transform(std::cbegin(ptrList), std::cend(ptrList), std::back_inserter(result), [](const AccountStatePtr &account) {
+            return account.data();
+        });
+        return result;
+    }();
+    FolderMan *folderMan = FolderMan::instance();
+    const auto folders = folderMan->map();
+    for (auto f : folders) {
```

Comment:
@kongomongo I would try to avoid duplicate code in `
void AccountSettings::slotForceSyncCurrentFolder()` and `void Systray::slotForceSyncAllFolders()` by moving the code for syncing the current folder into a `FolderMan` and allowing it to accept the argument of a `Folder *folderToSync`